app/models/driver.rb

```diff
@@ -0,0 +1,33 @@
+class Driver < ApplicationRecord
+  validates :name, presence: true
+  validates :vin, length: {is: 17}
+  # validates :available, presence: true
+
+  has_many :trips  #, dependent: :destroy
+
+  def average_rating
+    all_ratings = self.trips.map { |trip| trip.rating}
+    return nil if all_ratings.empty?
+
+    number_ratings = all_ratings.filter { |rating| rating unless rating.nil?}
+    average = number_ratings.sum / number_ratings.length.to_f
+    return average / 10 == 0 ? average : average.round(1)
+  end
```

Comment:
This is such lovely and clean code! Great implementation. Well done!! I'm inspired!!!